File: newsletter/src/main/java/org/com/newsletter/auth/domain/repository/MemberRepository.java
package org.com.newsletter.auth.domain.repository;

import java.util.Optional;
import org.com.newsletter.auth.domain.Member;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface MemberRepository extends JpaRepository<Member, Long> {
    Optional<Member> findByUsername(String username);
    boolean existsByUsername(String username);
}

File: ssr-board/src/main/kotlin/org/com/ssrboard/repository/MemberRepository.kt
package org.com.ssrboard.repository

import org.com.ssrboard.domain.Member
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface MemberRepository : JpaRepository<Member, Long> {
    fun findByNickname(nickname: String): Member?
}